src/mdns.rs

```diff
@@ -3,64 +3,89 @@
 use crate::devices::SharedDevices;
 use log::info;
 use std::net::IpAddr;
-use std::sync::Arc;
+use std::any::Any;
+use std::sync::{Arc, Mutex as StdMut};
+use std::time::Duration;
 
 use tokio::sync::Mutex;
 
 #[cfg(not(feature = "zeroconf"))]
 use {
     futures_util::{pin_mut, stream::StreamExt},
     mdns::{Record, RecordKind},
-    std::time::Duration,
 };
 
 #[cfg(feature = "zeroconf")]
 use {
     zeroconf::prelude::*,
-    zeroconf::{MdnsBrowser, ServiceType},
+    zeroconf::{MdnsBrowser, ServiceDiscovery, ServiceType},
 };
 
 const SERVICE_NAME: &str = "apple-mobdev2";
 const SERVICE_PROTOCOL: &str = "tcp";
 
+#[derive(Default, Debug, Clone)]
+pub struct Context {
+    name: String,
+    address: String,
+}
+
+#[cfg(feature = "zeroconf")]
+fn on_service_discovered(result: zeroconf::Result<ServiceDiscovery>, _context: Option<Arc<dyn Any>>,) {
```

Comment:
Context shouldn't start with a _ as it's used

Reply:
Renamed variables.
commit 57b380b98e4b8552bf96a655921ad88fbd9a2268